src/server/plugins/engine/components/MonthYearField.test.ts

```diff
@@ -1,57 +1,423 @@
+          {
+            title: 'example date parts field',
+            name: 'myComponent',
+            type: ComponentType.MonthYearField,
+            options: { required: false }
+          },
+          formModel
+        )
+
+        const { formSchema } = componentOptional.children
+
+        expect(formSchema.describe().keys).toEqual(
+          expect.objectContaining({
+            myComponent__month: expect.objectContaining({
+              allow: ['']
+            }),
+            myComponent__year: expect.objectContaining({
+              allow: ['']
+            })
+          })
+        )
+
+        const result = formSchema.validate(
+          getFormData({
+            month: '',
+            year: ''
+          }),
+          opts
+        )
+
+        expect(result.error).toBeUndefined()
+      })
+
+      it('accepts valid values', () => {
+        const { formSchema } = component.children
+
+        const result1 = formSchema.validate(
+          getFormData({
+            month: '12',
+            year: '2024'
+          }),
+          opts
+        )
+
+        const result2 = formSchema.validate(
+          getFormData({
+            month: '2',
+            year: '2024'
+          }),
+          opts
+        )
+
+        expect(result1.error).toBeUndefined()
+        expect(result2.error).toBeUndefined()
+      })
+
+      it('adds errors for empty value', () => {
+        const { formSchema } = component.children
+
+        const result = formSchema.validate(
+          getFormData({
+            month: '',
+            year: ''
+          }),
+          opts
+        )
+
+        expect(result.error).toEqual(
+          expect.objectContaining({
+            message: 'month must be between 1 and 12. year must be a number'
+          })
+        )
+      })
+
+      it('adds errors for invalid values', () => {
+        const { formSchema } = component.children
+
+        const result1 = formSchema.validate(['invalid'], opts)
+        const result2 = formSchema.validate({ unknown: 'invalid' }, opts)
+        const result3 = formSchema.validate(
+          getFormData({
+            month: 'invalid',
+            year: 'invalid'
+          }),
+          opts
+        )
+
+        expect(result1.error).toBeTruthy()
+        expect(result2.error).toBeTruthy()
+        expect(result3.error).toBeTruthy()
+      })
+    })
+
+    describe('State', () => {
+      const date = new Date('2024-12-31')
+
+      it('returns text from state', () => {
+        const state = getFormState(date)
+        const text = component.getDisplayStringFromState(state)
+
+        expect(text).toBe('December 2024')
+      })
+
+      it.skip('returns payload from state', () => {
+        const state = getFormState(startOfDay(date))
+        const payload = component.getFormDataFromState(state)
+
+        // TODO: Fix empty form data
+        expect(payload).toEqual(getFormData(date))
+      })
```

Comment:
By adding test coverage we reveal [bug #452104](https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/452104)

Will fix in another PR